yt_dlp/extractor/bandcamp.py

```diff
@@ -429,3 +421,88 @@ def _real_extract(self, url):
                 ie=self._TYPE_IE_DICT[item[1]])
             for item in items]
         return self.playlist_result(entries, id)
+
+
+class BandcampUserIE(InfoExtractor):
+    IE_NAME = 'Bandcamp:user'
+    _VALID_URL = r'https?://(?:(?P<id>[^.]+)\.)?bandcamp\.com'
+
+    _TESTS = [{
+        # Type 1 Bandcamp user page.
+        'url': 'https://adrianvonziegler.bandcamp.com',
+        'info_dict': {
+            'id': 'adrianvonziegler',
+            'title': 'Discography of adrianvonziegler',
+        },
+        'playlist_mincount': 23,
+    }, {
+        # Bandcamp user page with only one album
+        'url': 'http://dotscale.bandcamp.com',
+        'info_dict': {'id': 'dotscale', 'title': 'Discography of dotscale'},
+        'playlist_count': 1,
+    }, {
+        # Type 2 Bandcamp user page.
+        'url': 'https://nightcallofficial.bandcamp.com',
+        'info_dict': {
+            'id': 'nightcallofficial',
+            'title': 'Discography of nightcallofficial',
+        },
+        'playlist_count': 4,
+    }, {
+        'url': 'https://steviasphere.bandcamp.com/music',
+        'playlist_mincount': 47,
+        'info_dict': {
+            'id': 'steviasphere',
+            'title': 'Discography of steviasphere',
+        },
+    }, {
+        'url': 'https://coldworldofficial.bandcamp.com/music',
+        'playlist_mincount': 10,
+        'info_dict': {
+            'id': 'coldworldofficial',
+            'title': 'Discography of coldworldofficial',
+        },
+    }, {
+        'url': 'https://nuclearwarnowproductions.bandcamp.com/music',
+        'playlist_mincount': 399,
+        'info_dict': {
+            'id': 'nuclearwarnowproductions',
+            'title': 'Discography of nuclearwarnowproductions',
+        },
+    },
+    ]
+
+    @classmethod
+    def suitable(cls, url):
+        return (
+            False
+            if BandcampAlbumIE.suitable(url)
+            or BandcampIE.suitable(url)
+            or BandcampWeeklyIE.suitable(url)
+            else super(BandcampUserIE, cls).suitable(url))
```

Comment:
I noticed that this is seriously over capturing. Can you give example why you can't just capture the homepage and /music?

Reply:
I see you are having trouble with the regex. Just let me know whether these are the only URLs and I can fix it

> Can you give example why you can't just capture the homepage and /music?

Reply:
Thanks @pukkandan , I think I got it this time :)